narf/tfutils.py

```diff
@@ -1,47 +1,53 @@
 import tensorflow as tf
 
-def function_to_tflite(func, input_signature):
+def function_to_tflite(funcs, input_signatures):
     """Convert function to tflite model using python dynamic execution trickery to ensure that inputs
     and outputs are alphabetically ordered, since this is apparently the only way to prevent tflite from
     scrambling them"""
 
-    inputs = []
-    for i in range(len(input_signature)):
-        input_name = f"input_{i:05d}"
-        inputs.append(input_name)
+    if not isinstance(funcs, list):
+        funcs = [funcs]
+        input_signatures = [input_signatures]
 
-    def wrapped_func(*args):
-        outputs = func(*args)
+    def wrapped_func(iif, *args):
+        outputs = funcs[iif](*args)
 
         if not isinstance(outputs, tuple):
             outputs = (outputs,)
 
         output_dict = {}
         for i,output in enumerate(outputs):
-            output_name = f"output_{i:05d}"
+            output_name = f"output_{iif:05d}_{i:05d}"
             output_dict[output_name] = output
 
         return output_dict
 
-    arg_string = ", ".join(inputs)
+    arg_string = []
+    for iif, input_signature in enumerate(input_signatures):
+        inputs = []
+        for i in range(len(input_signature)):
+            input_name = f"input_{iif:05d}_{i:05d}"
+            inputs.append(input_name)
+        arg_string.append(", ".join(inputs))
 
     def_string = ""
-    def_string += "def make_module(wrapped_func, input_signature):\n"
+    def_string += "def make_module(wrapped_func, input_signatures):\n"
     def_string += "    class Export_Module(tf.Module):\n"
-    def_string += "        @tf.function(input_signature = input_signature)\n"
-    def_string += f"        def __call__(self, {arg_string}):\n"
-    def_string += f"            return wrapped_func({arg_string})\n"
+    for i, func in enumerate(funcs):
+        def_string += f"        @tf.function(input_signature = input_signatures[{i}])\n"
+        def_string += f"        def {func.__name__}(self, {arg_string[i]}):\n"
```

Comment:
So this works correctly if ```arg_string[i]``` is an empty string?

Reply:
Yes I tested and it works fine